### ISSUE TITLE
Incorrect type in cloudformation GetTemplateOutputTypeDef

### ISSUE THREAD

**Issue by mwgamble** [youtype/mypy_boto3_builder#174]
**Describe the bug**
The `TemplateBody` parameter in `mypy_boto3_cloudformation.type_defs.GetTemplateOutputTypeDef` is currently typed as `str`. This is incorrect - the template is returned as an `OrderedDict` (that itself contains nested lists and `OrderedDict`s).

I see this API is explicitly referred to here: https://github.com/youtype/mypy_boto3_builder/blob/a02f73ec5dc433e65bba44c2a08db688ac3d4762/mypy_boto3_builder/botocore_stubs_static/handlers.pyi#L77

But I don't know if that's in any way relevant.

**To Reproduce**

1. Install `boto3-stubs[...]`
2. Run `mypy`/`pyright` on the following code sample

```python
import boto3
from mypy_boto3_cloudformation.client import CloudFormationClient
from typing_extensions import reveal_type

cfn: CloudFormationClient = boto3.client("cloudformation")
response = cfn.get_template(StackName="copilot-vpc-service")
template = response["TemplateBody"]

reveal_type(template)
```

**Actual output**

```
❯ mypy test.py
test.py:9: note: Revealed type is "builtins.str"
Success: no issues found in 1 source file
```

**Expected output**

```
❯ mypy test.py
test.py:9: note: Revealed type is "collections.OrderedDict[str, Any]"
Success: no issues found in 1 source file
```

**Additional context**
Arch Linux, Python 3.10.8.

Record of installed dependencies inside a freshly created virtualenv:

```
❯ pip --version
pip 22.3.1 from /tmp/boto-stubs/venv/lib/python3.10/site-packages/pip (python 3.10)

❯ pip install mypy boto3 'boto3-stubs[cloudformation]' typing-extensions
Collecting mypy
  Using cached mypy-0.991-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (18.2 MB)
Collecting boto3
  Using cached boto3-1.26.34-py3-none-any.whl (132 kB)
Collecting boto3-stubs[cloudformation]
  Using cached boto3_stubs-1.26.34-py3-none-any.whl (50 kB)
Collecting typing-extensions
  Using cached typing_extensions-4.4.0-py3-none-any.whl (26 kB)
Collecting mypy-extensions>=0.4.3
  Using cached mypy_extensions-0.4.3-py2.py3-none-any.whl (4.5 kB)
Collecting tomli>=1.1.0
  Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting botocore<1.30.0,>=1.29.34
  Using cached botocore-1.29.34-py3-none-any.whl (10.3 MB)
Collecting s3transfer<0.7.0,>=0.6.0
  Using cached s3transfer-0.6.0-py3-none-any.whl (79 kB)
Collecting jmespath<2.0.0,>=0.7.1
  Using cached jmespath-1.0.1-py3-none-any.whl (20 kB)
Collecting botocore-stubs
  Using cached botocore_stubs-1.29.34-py3-none-any.whl (46 kB)
Collecting types-s3transfer
  Using cached types_s3transfer-0.6.0.post5-py3-none-any.whl (17 kB)
Collecting mypy-boto3-cloudformation<1.27.0,>=1.26.0
  Using cached mypy_boto3_cloudformation-1.26.11.post1-py3-none-any.whl (64 kB)
Collecting python-dateutil<3.0.0,>=2.1
  Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
Collecting urllib3<1.27,>=1.25.4
  Using cached urllib3-1.26.13-py2.py3-none-any.whl (140 kB)
Collecting types-awscrt
  Using cached types_awscrt-0.16.1-py3-none-any.whl (15 kB)
Collecting six>=1.5
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Installing collected packages: mypy-extensions, urllib3, typing-extensions, types-awscrt, tomli, six, jmespath, types-s3transfer, python-dateutil, mypy-boto3-cloudformation, mypy, botocore-stubs, botocore, boto3-stubs, s3transfer, boto3
Successfully installed boto3-1.26.34 boto3-stubs-1.26.34 botocore-1.29.34 botocore-stubs-1.29.34 jmespath-1.0.1 mypy-0.991 mypy-boto3-cloudformation-1.26.11.post1 mypy-extensions-0.4.3 python-dateutil-2.8.2 s3transfer-0.6.0 six-1.16.0 tomli-2.0.1 types-awscrt-0.16.1 types-s3transfer-0.6.0.post5 typing-extensions-4.4.0 urllib3-1.26.13
```

**Comment by mwgamble** [comment]
Though of course instead of an `OrderedDict`, it should probably be `Mapping`.

**Comment by mwgamble** [comment]
It may be worth creating a more customised `TypedDict` that includes all of the known top-level keys for a template, such as `Resources`, `Outputs`, etc.

**Comment by vemel** [comment]
Thank you for the report!

As I see, auto-generated boto3 docs are correct: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/cloudformation.html#CloudFormation.Client.get_template

So it might be an issue with my generator, or probably some kind of runtime injection from `botocore`. I will try to fix it today.

**Comment by vemel** [comment]
Yes, `botocore` returns `TemplateBody` processed with `json_decode_template_body`, even though botocore schemas reference the same `TemplateBody` StringShape. Fixed in `mypy-boto3-cloudformation 1.26.35`

Please update and let me know if the fix works as it should for you!

**Comment by mwgamble** [comment]
@vemel Wow, thanks for such a fast response! I'm on Christmas break now so it'll probably be a couple of weeks before I get a chance to test. Thank you though :)
